src/forms/create-alert.ts

```diff
@@ -1,38 +1,45 @@
-import { AlertCreate, AlertResponderType, AppCallRequest, AppCallValues, Identifier, IdentifierType } from '../types';
+import { AlertCreate, AlertResponderType, AppActingUser, AppCallRequest, AppCallValues, Identifier, IdentifierType, Team } from '../types';
 import { OpsGenieClient, OpsGenieOptions } from '../clients/opsgenie';
-import { ConfigStoreProps, KVStoreClient, KVStoreOptions } from '../clients/kvstore';
-import { AlertCreateForm, ExceptionType, StoreKeys, option_alert_priority_p3 } from '../constant';
+import { AlertCreateForm, ExceptionType, option_alert_priority_p3 } from '../constant';
 import { configureI18n } from '../utils/translations';
-import { tryPromise } from '../utils/utils';
+import { isUserSystemAdmin, tryPromise } from '../utils/utils';
+import { allowMemberAction, getOpsGenieAPIKey, validateUserAccess } from '../utils/user-mapping';
+import { Exception } from '../utils/exception';
 
 export async function createAlertCall(call: AppCallRequest): Promise<string> {
-    const mattermostUrl: string | undefined = call.context.mattermost_site_url;
-    const botAccessToken: string | undefined = call.context.bot_access_token;
+    const actingUser: AppActingUser | undefined = call.context.acting_user;
+    const isSystemAdmin: boolean = isUserSystemAdmin(actingUser);
+    const allowMember: boolean = allowMemberAction(call.context);
     const values: AppCallValues | undefined = call.values;
+    const apiKey = getOpsGenieAPIKey(call);
     const i18nObj = configureI18n(call.context);
 
     const message: string = values?.[AlertCreateForm.ALERT_MESSAGE];
     const priority: string = values?.[AlertCreateForm.ALERT_PRIORITY]?.value || option_alert_priority_p3;
     const teamName: string = values?.[AlertCreateForm.TEAM_NAME];
 
-    const options: KVStoreOptions = {
-        mattermostUrl: <string>mattermostUrl,
-        accessToken: <string>botAccessToken,
-    };
-    const kvStoreClient = new KVStoreClient(options);
-
-    const config: ConfigStoreProps = await kvStoreClient.kvGet(StoreKeys.config);
-
     const optionsOpsgenie: OpsGenieOptions = {
-        api_key: config.opsgenie_apikey,
+        api_key: apiKey,
     };
     const opsGenieClient = new OpsGenieClient(optionsOpsgenie);
 
     const identifier: Identifier = {
         identifier: teamName,
         identifierType: IdentifierType.NAME,
     };
-    await tryPromise(opsGenieClient.getTeam(identifier), ExceptionType.MARKDOWN, i18nObj.__('forms.error'));
+
+    const team: Team = await tryPromise<Team>(opsGenieClient.getTeam(identifier), ExceptionType.MARKDOWN, i18nObj.__('forms.error'));
+
+    if (!allowMember) {
+        throw new Exception(ExceptionType.MARKDOWN, i18nObj.__('general.validation-user.genie-action-invalid'));
+    }
+
+    if (!isSystemAdmin) {
+        const teamMembers: string[] | undefined = team?.members?.map((member) => member.user.username);
+        if (!teamMembers || !teamMembers.includes(actingUser?.email || '')) {
```

Comment:
`!teamMembers.includes('')` doesn't seem like a useful check. Will that ever be the case?

Reply:
This was solved in commit [d91fc61](https://github.com/liz-segura98/mattermost-app-opsgenie/pull/1/commits/d91fc61f3d5c4e7a5418a74cce2c5a72c991a951)